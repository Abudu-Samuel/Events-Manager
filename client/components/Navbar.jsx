import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <div>
    <header className="header">
      <nav className="navbar navbar-expand-lg navbar-dark fixed-top">
        <div className="container">
          <Link to="/" className="navbar-brand">Events Manager</Link>
          <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation"><span className="navbar-toggler-icon" /></button>
          <div className="collapse navbar-collapse" id="navbarSupportedContent">
            <ul className="navbar-nav mr-auto smooth-scroll">
              <li className="nav-item">
                <Link to="/" className="nav-link">Home</Link>
              </li>
            </ul>
            <ul className="navbar-nav ml-auto nav-flex-icons">
              <li className="nav-item">
                <a className="nav-link waves-effect waves-light"><i className="fa fa-twitter" /></a>
              </li>
              <li className="nav-item">
                <a className="nav-link waves-effect waves-light"><i className="fa fa-google-plus" /></a>
              </li>
              <li className="nav-item">
                <Link to="/signup" className="nav-link waves-effect waves-light">Sign Up</Link>
              </li>
              <li className="nav-item">
                <Link to="/signin" className="nav-link waves-effect waves-light">Sign In</Link>
              </li>
              <li className="nav-item">
                <Link to="/addcenter" className="nav-link waves-effect waves-light">Admin</Link>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </header>
  </div>
);

export default Navbar;
