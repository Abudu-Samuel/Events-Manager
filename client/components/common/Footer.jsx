import React from 'react';

const Footer = () => (
  <div>
    <footer className="page-footer center-on-small-only default-color pt-0">
      <div className="footer-copyright">
        <div className="container-fluid">
                Â© 2017 Copyright: <a href="#"><strong> Events Manager.com
          </strong></a>
        </div>
      </div>
    </footer>
  </div>
);

export default Footer;
