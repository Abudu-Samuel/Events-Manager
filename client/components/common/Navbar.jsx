import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import * as userActions from '../../actions/actionCreator';
import history from '../../history';

/**
 * @class Navbar
 *
 * @extends {React.Component}
 */
export class Navbar extends React.Component {
  /**
   * Creates an instance of Navbar.
   *
   * @param {any} props
   *
   * @memberof Navbar
   */
  constructor(props) {
    super(props);
    this.logOut = this.logOut.bind(this);
  }

  /**
   * @method logOut
   *
   * @description handle log out
   *
   * @returns {string} cleared storage
   *
   * @param {event} event
   *
   * @memberof Navbar
   */
  logOut(event) {
    event.preventDefault();
    localStorage.removeItem('x-access-token');
    this.props.logOut();
    history.push('/');
  }

  /**
   * @method render
   *
   * @returns {jsx} Jsx representation of the dom
   *
   * @description React render method
   *
   * @memberof Navbar
   */
  render() {
    const { isAuthenticated, userData } = this.props;
    return (
      <div>
        <header className="header">
          <nav className="navbar z-depth-2 mb-4 navbar-expand-lg
           navbar-dark fixed-top">
            <div className="container">
              <Link to="/" className="font-weight-bold navbar-brand">
              Events Manager</Link>
              <button className="navbar-toggler" type="button"
                data-toggle="collapse" data-target="#navbarSupportedContent"
                aria-controls="navbarSupportedContent" aria-expanded="false"
                aria-label="Toggle navigation">
                <span className="navbar-toggler-icon" /></button>
              <div className="collapse navbar-collapse"
                id="navbarSupportedContent">
                <ul className="navbar-nav mr-auto smooth-scroll">
                  <li className="nav-item">
                    <Link to="/" className="font-weight-bold nav-link">Home</Link>
                  </li>
                </ul>
                {
                  isAuthenticated ? <ul className="navbar-nav ml-auto
                   nav-flex-icons">
                    <li className="nav-item dropdown">
                      <Link className="font-weight-bold nav-link dropdown-toggle
                       waves-effect waves-light"
                      id="navbarDropdownMenuLink" name="dash-drop" data-toggle="dropdown"
                      aria-haspopup="true"
                      aria-expanded="false" to='/addcenter'>
                        <i className="fa fa-user mr-2" /><span>Hi </span>
                        {
                          userData.username.split('')[0].toUpperCase()
                           + userData.username.slice(1)
                        }
                      </Link>
                      <div className="dropdown-menu dropdown-menu-right
                       dropdown-unique" aria-labelledby="navbarDropdownMenuLink">
                        {
                          userData.isAdmin ?
                            <div>
                              <Link className="font-weight-bold dropdown-item
                               waves-effect btn btn-size waves-light ml-1
                                text-center" to="/admin/dashboard">Dashboard</Link>
                              <Link className="font-weight-bold dropdown-item
                               waves-effect btn btn-size waves-light ml-1
                                text-center" to="/addcenter">Add Center</Link>
                              <Link className="font-weight-bold dropdown-item
                               waves-effect btn btn-size waves-light ml-1
                                text-center" to="/manage/centers">View Centers</Link>
                            </div>
                            :
                            <div>
                              <Link className="font-weight-bold dropdown-item
                               waves-effect btn btn-size waves-light ml-1
                               text-center" to="/dashboard">Dashboard</Link>
                              <Link className="font-weight-bold dropdown-item
                              waves-effect manage-event btn btn-size waves-light ml-1
                               text-center" to="/manage/events">Manage Event</Link>
                            </div>
                        }
                        <button onClick={this.logOut}
                          className="font-weight-bold dropdown-item waves-effect
                         btn btn-mycolor btn-size waves-light ml-1 text-center">
                         Log Out</button>
                      </div>
                    </li>
                  </ul> : <ul className="navbar-nav ml-auto nav-flex-icons">
                    <li className="nav-item">
                      <Link to="/signup" className="font-weight-bold nav-link
                       waves-effect waves-light-signup">Sign Up</Link>
                    </li>
                    <li className="nav-item">
                      <Link to="/signin" className="font-weight-bold nav-link
                       waves-effect waves-light-signin">Sign In</Link>
                    </li>
                  </ul>
                }
              </div>
            </div>
          </nav>
        </header>
      </div>
    );
  }
}

Navbar.propTypes = {
  logOut: PropTypes.func,
  userData: PropTypes.object,
  isAuthenticated: PropTypes.bool
};

/**
 * @description Redux connect parameter - mapDispatchToProps
 *
 * @param {function} state
 *
 * @return {object} mapped dispatch
 */
const mapStateToProps = state => ({
  isAuthenticated: state.userAccess.isAuthenticated,
  userData: state.userAccess.userData,
});

/**
 * @description Redux connect parameter - mapDispatchToProps
 *
 * @param {function} dispatch
 *
 * @return {object} mapped dispatch
 */

export const mapDispatchToProps = dispatch => bindActionCreators({
  logOut: () => userActions.logout()
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Navbar);
