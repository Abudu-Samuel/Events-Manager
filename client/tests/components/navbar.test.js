import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { Navbar, mapDispatchToProps } from '../../components/common/Navbar';

const logOut = {
  preventDefault: jest.fn(),
};

const props = {
  logOut: () => Promise.resolve()
};

describe('create Component', () => {
  it('SHOULD RENDER COMPONENT CORRECTLY', () => {
    const wrapper = shallow(<Navbar />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('SHOULD RENDER COMPONENT CORRECTLY', () => {
    const wrapper = shallow(<Navbar {...props}/>);
    wrapper.instance().logOut(logOut);
    expect(logOut.preventDefault).toHaveBeenCalled();
  });

  it('should return an object of functions to be passed as props', () => {
    const dispatchFn = jest.fn();
    const propsObj = mapDispatchToProps(dispatchFn);

    expect(propsObj).toEqual({
      logOut: expect.any(Function)
    });

    propsObj.logOut();
    expect(dispatchFn).toHaveBeenCalledTimes(1);
  });
});
