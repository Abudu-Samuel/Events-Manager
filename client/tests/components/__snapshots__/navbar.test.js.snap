// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create Component SHOULD RENDER COMPONENT CORRECTLY 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <header
        className="header"
      >
        <nav
          className="navbar z-depth-2 mb-4 navbar-expand-lg navbar-dark fixed-top"
        >
          <div
            className="container"
          >
            <Link
              className="font-weight-bold navbar-brand"
              replace={false}
              to="/"
            >
              Events Manager
            </Link>
            <button
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
              className="navbar-toggler"
              data-target="#navbarSupportedContent"
              data-toggle="collapse"
              type="button"
            >
              <span
                className="navbar-toggler-icon"
              />
            </button>
            <div
              className="collapse navbar-collapse"
              id="navbarSupportedContent"
            >
              <ul
                className="navbar-nav mr-auto smooth-scroll"
              >
                <li
                  className="nav-item"
                >
                  <Link
                    className="font-weight-bold nav-link"
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>
                </li>
              </ul>
              <ul
                className="navbar-nav ml-auto nav-flex-icons"
              >
                <li
                  className="nav-item"
                >
                  <Link
                    className="font-weight-bold nav-link waves-effect waves-light-signup"
                    replace={false}
                    to="/signup"
                  >
                    Sign Up
                  </Link>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    className="font-weight-bold nav-link waves-effect waves-light-signin"
                    replace={false}
                    to="/signin"
                  >
                    Sign In
                  </Link>
                </li>
              </ul>
            </div>
          </div>
        </nav>
      </header>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav
          className="navbar z-depth-2 mb-4 navbar-expand-lg navbar-dark fixed-top"
        >
          <div
            className="container"
          >
            <Link
              className="font-weight-bold navbar-brand"
              replace={false}
              to="/"
            >
              Events Manager
            </Link>
            <button
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
              className="navbar-toggler"
              data-target="#navbarSupportedContent"
              data-toggle="collapse"
              type="button"
            >
              <span
                className="navbar-toggler-icon"
              />
            </button>
            <div
              className="collapse navbar-collapse"
              id="navbarSupportedContent"
            >
              <ul
                className="navbar-nav mr-auto smooth-scroll"
              >
                <li
                  className="nav-item"
                >
                  <Link
                    className="font-weight-bold nav-link"
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>
                </li>
              </ul>
              <ul
                className="navbar-nav ml-auto nav-flex-icons"
              >
                <li
                  className="nav-item"
                >
                  <Link
                    className="font-weight-bold nav-link waves-effect waves-light-signup"
                    replace={false}
                    to="/signup"
                  >
                    Sign Up
                  </Link>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    className="font-weight-bold nav-link waves-effect waves-light-signin"
                    replace={false}
                    to="/signin"
                  >
                    Sign In
                  </Link>
                </li>
              </ul>
            </div>
          </div>
        </nav>,
        "className": "header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <Link
              className="font-weight-bold navbar-brand"
              replace={false}
              to="/"
            >
              Events Manager
            </Link>
            <button
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
              className="navbar-toggler"
              data-target="#navbarSupportedContent"
              data-toggle="collapse"
              type="button"
            >
              <span
                className="navbar-toggler-icon"
              />
            </button>
            <div
              className="collapse navbar-collapse"
              id="navbarSupportedContent"
            >
              <ul
                className="navbar-nav mr-auto smooth-scroll"
              >
                <li
                  className="nav-item"
                >
                  <Link
                    className="font-weight-bold nav-link"
                    replace={false}
                    to="/"
                  >
                    Home
                  </Link>
                </li>
              </ul>
              <ul
                className="navbar-nav ml-auto nav-flex-icons"
              >
                <li
                  className="nav-item"
                >
                  <Link
                    className="font-weight-bold nav-link waves-effect waves-light-signup"
                    replace={false}
                    to="/signup"
                  >
                    Sign Up
                  </Link>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    className="font-weight-bold nav-link waves-effect waves-light-signin"
                    replace={false}
                    to="/signin"
                  >
                    Sign In
                  </Link>
                </li>
              </ul>
            </div>
          </div>,
          "className": "navbar z-depth-2 mb-4 navbar-expand-lg navbar-dark fixed-top",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="font-weight-bold navbar-brand"
                replace={false}
                to="/"
              >
                Events Manager
              </Link>,
              <button
                aria-controls="navbarSupportedContent"
                aria-expanded="false"
                aria-label="Toggle navigation"
                className="navbar-toggler"
                data-target="#navbarSupportedContent"
                data-toggle="collapse"
                type="button"
              >
                <span
                  className="navbar-toggler-icon"
                />
              </button>,
              <div
                className="collapse navbar-collapse"
                id="navbarSupportedContent"
              >
                <ul
                  className="navbar-nav mr-auto smooth-scroll"
                >
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="font-weight-bold nav-link"
                      replace={false}
                      to="/"
                    >
                      Home
                    </Link>
                  </li>
                </ul>
                <ul
                  className="navbar-nav ml-auto nav-flex-icons"
                >
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="font-weight-bold nav-link waves-effect waves-light-signup"
                      replace={false}
                      to="/signup"
                    >
                      Sign Up
                    </Link>
                  </li>
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="font-weight-bold nav-link waves-effect waves-light-signin"
                      replace={false}
                      to="/signin"
                    >
                      Sign In
                    </Link>
                  </li>
                </ul>
              </div>,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Events Manager",
                "className": "font-weight-bold navbar-brand",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Events Manager",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-controls": "navbarSupportedContent",
                "aria-expanded": "false",
                "aria-label": "Toggle navigation",
                "children": <span
                  className="navbar-toggler-icon"
                />,
                "className": "navbar-toggler",
                "data-target": "#navbarSupportedContent",
                "data-toggle": "collapse",
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "navbar-toggler-icon",
                },
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <ul
                    className="navbar-nav mr-auto smooth-scroll"
                  >
                    <li
                      className="nav-item"
                    >
                      <Link
                        className="font-weight-bold nav-link"
                        replace={false}
                        to="/"
                      >
                        Home
                      </Link>
                    </li>
                  </ul>,
                  <ul
                    className="navbar-nav ml-auto nav-flex-icons"
                  >
                    <li
                      className="nav-item"
                    >
                      <Link
                        className="font-weight-bold nav-link waves-effect waves-light-signup"
                        replace={false}
                        to="/signup"
                      >
                        Sign Up
                      </Link>
                    </li>
                    <li
                      className="nav-item"
                    >
                      <Link
                        className="font-weight-bold nav-link waves-effect waves-light-signin"
                        replace={false}
                        to="/signin"
                      >
                        Sign In
                      </Link>
                    </li>
                  </ul>,
                ],
                "className": "collapse navbar-collapse",
                "id": "navbarSupportedContent",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <li
                      className="nav-item"
                    >
                      <Link
                        className="font-weight-bold nav-link"
                        replace={false}
                        to="/"
                      >
                        Home
                      </Link>
                    </li>,
                    "className": "navbar-nav mr-auto smooth-scroll",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        className="font-weight-bold nav-link"
                        replace={false}
                        to="/"
                      >
                        Home
                      </Link>,
                      "className": "nav-item",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Home",
                        "className": "font-weight-bold nav-link",
                        "replace": false,
                        "to": "/",
                      },
                      "ref": null,
                      "rendered": "Home",
                      "type": [Function],
                    },
                    "type": "li",
                  },
                  "type": "ul",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <li
                        className="nav-item"
                      >
                        <Link
                          className="font-weight-bold nav-link waves-effect waves-light-signup"
                          replace={false}
                          to="/signup"
                        >
                          Sign Up
                        </Link>
                      </li>,
                      <li
                        className="nav-item"
                      >
                        <Link
                          className="font-weight-bold nav-link waves-effect waves-light-signin"
                          replace={false}
                          to="/signin"
                        >
                          Sign In
                        </Link>
                      </li>,
                    ],
                    "className": "navbar-nav ml-auto nav-flex-icons",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Link
                          className="font-weight-bold nav-link waves-effect waves-light-signup"
                          replace={false}
                          to="/signup"
                        >
                          Sign Up
                        </Link>,
                        "className": "nav-item",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Sign Up",
                          "className": "font-weight-bold nav-link waves-effect waves-light-signup",
                          "replace": false,
                          "to": "/signup",
                        },
                        "ref": null,
                        "rendered": "Sign Up",
                        "type": [Function],
                      },
                      "type": "li",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Link
                          className="font-weight-bold nav-link waves-effect waves-light-signin"
                          replace={false}
                          to="/signin"
                        >
                          Sign In
                        </Link>,
                        "className": "nav-item",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Sign In",
                          "className": "font-weight-bold nav-link waves-effect waves-light-signin",
                          "replace": false,
                          "to": "/signin",
                        },
                        "ref": null,
                        "rendered": "Sign In",
                        "type": [Function],
                      },
                      "type": "li",
                    },
                  ],
                  "type": "ul",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "nav",
      },
      "type": "header",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <header
          className="header"
        >
          <nav
            className="navbar z-depth-2 mb-4 navbar-expand-lg navbar-dark fixed-top"
          >
            <div
              className="container"
            >
              <Link
                className="font-weight-bold navbar-brand"
                replace={false}
                to="/"
              >
                Events Manager
              </Link>
              <button
                aria-controls="navbarSupportedContent"
                aria-expanded="false"
                aria-label="Toggle navigation"
                className="navbar-toggler"
                data-target="#navbarSupportedContent"
                data-toggle="collapse"
                type="button"
              >
                <span
                  className="navbar-toggler-icon"
                />
              </button>
              <div
                className="collapse navbar-collapse"
                id="navbarSupportedContent"
              >
                <ul
                  className="navbar-nav mr-auto smooth-scroll"
                >
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="font-weight-bold nav-link"
                      replace={false}
                      to="/"
                    >
                      Home
                    </Link>
                  </li>
                </ul>
                <ul
                  className="navbar-nav ml-auto nav-flex-icons"
                >
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="font-weight-bold nav-link waves-effect waves-light-signup"
                      replace={false}
                      to="/signup"
                    >
                      Sign Up
                    </Link>
                  </li>
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="font-weight-bold nav-link waves-effect waves-light-signin"
                      replace={false}
                      to="/signin"
                    >
                      Sign In
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </nav>
        </header>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <nav
            className="navbar z-depth-2 mb-4 navbar-expand-lg navbar-dark fixed-top"
          >
            <div
              className="container"
            >
              <Link
                className="font-weight-bold navbar-brand"
                replace={false}
                to="/"
              >
                Events Manager
              </Link>
              <button
                aria-controls="navbarSupportedContent"
                aria-expanded="false"
                aria-label="Toggle navigation"
                className="navbar-toggler"
                data-target="#navbarSupportedContent"
                data-toggle="collapse"
                type="button"
              >
                <span
                  className="navbar-toggler-icon"
                />
              </button>
              <div
                className="collapse navbar-collapse"
                id="navbarSupportedContent"
              >
                <ul
                  className="navbar-nav mr-auto smooth-scroll"
                >
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="font-weight-bold nav-link"
                      replace={false}
                      to="/"
                    >
                      Home
                    </Link>
                  </li>
                </ul>
                <ul
                  className="navbar-nav ml-auto nav-flex-icons"
                >
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="font-weight-bold nav-link waves-effect waves-light-signup"
                      replace={false}
                      to="/signup"
                    >
                      Sign Up
                    </Link>
                  </li>
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="font-weight-bold nav-link waves-effect waves-light-signin"
                      replace={false}
                      to="/signin"
                    >
                      Sign In
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </nav>,
          "className": "header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
            >
              <Link
                className="font-weight-bold navbar-brand"
                replace={false}
                to="/"
              >
                Events Manager
              </Link>
              <button
                aria-controls="navbarSupportedContent"
                aria-expanded="false"
                aria-label="Toggle navigation"
                className="navbar-toggler"
                data-target="#navbarSupportedContent"
                data-toggle="collapse"
                type="button"
              >
                <span
                  className="navbar-toggler-icon"
                />
              </button>
              <div
                className="collapse navbar-collapse"
                id="navbarSupportedContent"
              >
                <ul
                  className="navbar-nav mr-auto smooth-scroll"
                >
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="font-weight-bold nav-link"
                      replace={false}
                      to="/"
                    >
                      Home
                    </Link>
                  </li>
                </ul>
                <ul
                  className="navbar-nav ml-auto nav-flex-icons"
                >
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="font-weight-bold nav-link waves-effect waves-light-signup"
                      replace={false}
                      to="/signup"
                    >
                      Sign Up
                    </Link>
                  </li>
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="font-weight-bold nav-link waves-effect waves-light-signin"
                      replace={false}
                      to="/signin"
                    >
                      Sign In
                    </Link>
                  </li>
                </ul>
              </div>
            </div>,
            "className": "navbar z-depth-2 mb-4 navbar-expand-lg navbar-dark fixed-top",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  className="font-weight-bold navbar-brand"
                  replace={false}
                  to="/"
                >
                  Events Manager
                </Link>,
                <button
                  aria-controls="navbarSupportedContent"
                  aria-expanded="false"
                  aria-label="Toggle navigation"
                  className="navbar-toggler"
                  data-target="#navbarSupportedContent"
                  data-toggle="collapse"
                  type="button"
                >
                  <span
                    className="navbar-toggler-icon"
                  />
                </button>,
                <div
                  className="collapse navbar-collapse"
                  id="navbarSupportedContent"
                >
                  <ul
                    className="navbar-nav mr-auto smooth-scroll"
                  >
                    <li
                      className="nav-item"
                    >
                      <Link
                        className="font-weight-bold nav-link"
                        replace={false}
                        to="/"
                      >
                        Home
                      </Link>
                    </li>
                  </ul>
                  <ul
                    className="navbar-nav ml-auto nav-flex-icons"
                  >
                    <li
                      className="nav-item"
                    >
                      <Link
                        className="font-weight-bold nav-link waves-effect waves-light-signup"
                        replace={false}
                        to="/signup"
                      >
                        Sign Up
                      </Link>
                    </li>
                    <li
                      className="nav-item"
                    >
                      <Link
                        className="font-weight-bold nav-link waves-effect waves-light-signin"
                        replace={false}
                        to="/signin"
                      >
                        Sign In
                      </Link>
                    </li>
                  </ul>
                </div>,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Events Manager",
                  "className": "font-weight-bold navbar-brand",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Events Manager",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-controls": "navbarSupportedContent",
                  "aria-expanded": "false",
                  "aria-label": "Toggle navigation",
                  "children": <span
                    className="navbar-toggler-icon"
                  />,
                  "className": "navbar-toggler",
                  "data-target": "#navbarSupportedContent",
                  "data-toggle": "collapse",
                  "type": "button",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "navbar-toggler-icon",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <ul
                      className="navbar-nav mr-auto smooth-scroll"
                    >
                      <li
                        className="nav-item"
                      >
                        <Link
                          className="font-weight-bold nav-link"
                          replace={false}
                          to="/"
                        >
                          Home
                        </Link>
                      </li>
                    </ul>,
                    <ul
                      className="navbar-nav ml-auto nav-flex-icons"
                    >
                      <li
                        className="nav-item"
                      >
                        <Link
                          className="font-weight-bold nav-link waves-effect waves-light-signup"
                          replace={false}
                          to="/signup"
                        >
                          Sign Up
                        </Link>
                      </li>
                      <li
                        className="nav-item"
                      >
                        <Link
                          className="font-weight-bold nav-link waves-effect waves-light-signin"
                          replace={false}
                          to="/signin"
                        >
                          Sign In
                        </Link>
                      </li>
                    </ul>,
                  ],
                  "className": "collapse navbar-collapse",
                  "id": "navbarSupportedContent",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <li
                        className="nav-item"
                      >
                        <Link
                          className="font-weight-bold nav-link"
                          replace={false}
                          to="/"
                        >
                          Home
                        </Link>
                      </li>,
                      "className": "navbar-nav mr-auto smooth-scroll",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Link
                          className="font-weight-bold nav-link"
                          replace={false}
                          to="/"
                        >
                          Home
                        </Link>,
                        "className": "nav-item",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Home",
                          "className": "font-weight-bold nav-link",
                          "replace": false,
                          "to": "/",
                        },
                        "ref": null,
                        "rendered": "Home",
                        "type": [Function],
                      },
                      "type": "li",
                    },
                    "type": "ul",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <li
                          className="nav-item"
                        >
                          <Link
                            className="font-weight-bold nav-link waves-effect waves-light-signup"
                            replace={false}
                            to="/signup"
                          >
                            Sign Up
                          </Link>
                        </li>,
                        <li
                          className="nav-item"
                        >
                          <Link
                            className="font-weight-bold nav-link waves-effect waves-light-signin"
                            replace={false}
                            to="/signin"
                          >
                            Sign In
                          </Link>
                        </li>,
                      ],
                      "className": "navbar-nav ml-auto nav-flex-icons",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <Link
                            className="font-weight-bold nav-link waves-effect waves-light-signup"
                            replace={false}
                            to="/signup"
                          >
                            Sign Up
                          </Link>,
                          "className": "nav-item",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "children": "Sign Up",
                            "className": "font-weight-bold nav-link waves-effect waves-light-signup",
                            "replace": false,
                            "to": "/signup",
                          },
                          "ref": null,
                          "rendered": "Sign Up",
                          "type": [Function],
                        },
                        "type": "li",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <Link
                            className="font-weight-bold nav-link waves-effect waves-light-signin"
                            replace={false}
                            to="/signin"
                          >
                            Sign In
                          </Link>,
                          "className": "nav-item",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {
                            "children": "Sign In",
                            "className": "font-weight-bold nav-link waves-effect waves-light-signin",
                            "replace": false,
                            "to": "/signin",
                          },
                          "ref": null,
                          "rendered": "Sign In",
                          "type": [Function],
                        },
                        "type": "li",
                      },
                    ],
                    "type": "ul",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "nav",
        },
        "type": "header",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
